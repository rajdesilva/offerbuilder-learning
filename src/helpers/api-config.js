import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});