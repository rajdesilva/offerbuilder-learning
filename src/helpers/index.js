export * from './api-config';
export * from './history';
