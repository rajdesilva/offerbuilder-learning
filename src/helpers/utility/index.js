export { createSearchQueryURL } from './createSearchQueryURL';
export { isParentOrChildrenPropertyAdded } from './isParentOrChildrenPropertyAdded';
export { isPropertySelected } from './isPropertySelected';
export { mapBrands } from './mapBrands';
export { mapOnlySuppliers } from './mapOnlySuppliers';
export { parseFormDescriptionsAndImagesForOffer } from './parseFormDescriptionsAndImagesForOffer';
export { removePropertyForSelectedChannels } from './removePropertyForSelectedChannels';
export { removePropertyForSelectedSuppliers } from './removePropertyForSelectedSuppliers';
export { getParsedProperty } from './getParsedProperty';
export { getFormattedValue } from './getFormattedValue';
export { getValue } from './getValue';
export { parsePropertyMarketingInfoToFormData } from './parsePropertyMarketingInfoToFormData';
export { createOfferRequestObject } from './createOfferRequestObject';
export { getParsedOffers } from './getParsedOffers';
export { getListOfPropertiesForSupplier } from './getListOfPropertiesForSupplier';
export { getListOfPropertiesForChannels } from './getListOfPropertiesForChannels';
export { getInitializedState } from './getInitializedState';
export { getParsedNewOfferSettingsObject } from './getParsedNewOfferSettingsObject';
export { getListOfPropertiesForStorefront } from './getListOfPropertiesForStorefront';
export { checkIfMultipleChannelPresentForTarget } from './checkIfMultipleChannelPresentForTarget';
export { checkIfMultipleSupplierPresentForTarget } from './checkIfMultipleSupplierPresentForTarget';
export { checkIfMultipleStorefrontPresentForTarget } from './checkIfMultipleStorefrontPresentForTarget';
export { removePropertyForSelectedStorefront } from './removePropertyForSelectedStorefront';
export { checkIfMultipleBrandPresentForTarget } from './checkIfMultipleBrandPresentForTarget';
export { getListOfPropertiesForBrand } from './getListOfPropertiesForBrand';
export { getSearchStateFromRedux } from './getSearchStateFromRedux';
export { getObjectStringify } from './getObjectStringify';
export { parseOfferMarketingFromOffer } from './parseOfferMarketingFromOffer';
export { parseOfferMarketingLanguagesFromOffer } from './parseOfferMarketingLanguagesFromOffer';
export { getFilterData } from './getFilterData';
export { getSearchAndFilterQueryURL } from './getSearchAndFilterQueryURL';
export { getSearchQueryURLForEditAddProperties } from "./getSearchQueryURLForEditAddProperties";
export { isSavedOfferProperty } from './isSavedOfferProperty';
export { getParsedPropertiesForDeeplinkSettings } from "./getParsedPropertiesForDeeplinkSettings";
export { resetApplicationReduxAndStorage } from "./resetApplicationReduxAndStorage";
export { updateSettingsBrandWithSearchedValues } from "./updateSettingsBrandWithSearchedValues";
export { checkIfUserHasRole } from "./checkIfUserHasRole";
export { getUserOfferBuilderRole } from "./getUserOfferBuilderRole";
export { checkIfNewPropertiesAdded } from "./checkIfNewPropertiesAdded";
export { getAllBrandsWithoutKeysFromTempBrands } from "./getAllBrandsWithoutKeysFromTempBrands";
export { checkIfItemAlreadyPresentInList } from "./checkIfItemAlreadyPresentInList";
export { checkIfTravellingDateValidAsPerBookingDate } from "./checkIfTravellingDateValidAsPerBookingDate";
export { checkIfPropertiesHaveMultipleLocations } from "./checkIfPropertiesHaveMultipleLocations";
export { displayMargin } from "./displayMargin";
export { getFilterCount } from "./getFilterCount";
export { labelForBrands } from "./labelForBrands";
export { getDistanceFromLatLonInKm } from "./getDistanceFromLatLonInKm";
export { getTrustYouIdListFromProperties } from "./getTrustYouIdListFromProperties";
export { getSearchParamsToFetchMarketPrice } from "./getSearchParamsToFetchMarketPrice";
export { getMarketPriceForPropertyFromList } from "./getMarketPriceForPropertyFromList";
export { getImageFilterCount } from "./getImageFilterCount";
export { getSearchAndFilterImagesQueryUrl } from "./getSearchAndFilterImagesQueryUrl";
export { getImageName } from "./getImageName";
export { updateReduxWithOfferData } from "./updateReduxWithOfferData";
export { getOfferTypeFromPropertiesSelected } from "./getOfferTypeFromPropertiesSelected";