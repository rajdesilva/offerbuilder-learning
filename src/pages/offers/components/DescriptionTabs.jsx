import React from "react";
import { Tabs } from "antd";
import Descriptions from "./Descriptions";
import { useSelector } from "react-redux";
import { isEqual } from "lodash";
const { TabPane } = Tabs;

function DescriptionTabs(props) {
  const selectedLanguages = useSelector((state) =>
    window.getValue(state, "newoffermarketinginfo.selectedLanguages"),
    isEqual
  );
  const newOfferMarketingInfo = useSelector((state) =>
  window.getValue(state, "newoffermarketinginfo.marketingInfo"))
  return (
    <Tabs defaultActiveKey="EN" data-testid="description-tab" destroyInactiveTabPane>
      {selectedLanguages.map((language) => (
        <TabPane tab={language.name} key={language.id} forceRender destroyInactiveTabPane>
          <Descriptions
            data-testid="descriptions"
            language={language}
            propertyIndex={props.index}
            languageId={language.id}
            isForProperty={props.isForProperty}
            description={newOfferMarketingInfo[`${language.id}`]}
          />
        </TabPane>
      ))}
    </Tabs>
  );
}

export default DescriptionTabs;
