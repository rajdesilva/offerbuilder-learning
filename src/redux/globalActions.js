export const globalActions = {
  'SET_SUCCESS': 'SET_SUCCESS',
  'SET_ERROR': 'SET_ERROR',
  'SET_LANGUAGE': 'SET_LANGUAGE',
  'CLEAR': 'CLEAR',
}