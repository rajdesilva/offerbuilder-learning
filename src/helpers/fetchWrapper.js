import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
