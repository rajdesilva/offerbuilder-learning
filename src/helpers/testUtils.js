import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
