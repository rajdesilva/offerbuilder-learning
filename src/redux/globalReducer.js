import { globalActions } from "./globalActions";
import produce from "immer";

const initialState = {
  error: "",
  success: "",
  language: "",
};
export const globalReducer = produce((draft, action) => {
  switch (action.type) {
    case globalActions.SET_ERROR:
      draft.success = "";
      draft.error = action.payload;
      return;
    case globalActions.SET_SUCCESS:
      draft.error = "";
      draft.success = action.payload;
      return;
    case globalActions.SET_LANGUAGE:
      draft.language = localStorage.getItem('language')
      return;
    case globalActions.CLEAR:
      return initialState;
    default:
      return draft;
  }
}, initialState);
