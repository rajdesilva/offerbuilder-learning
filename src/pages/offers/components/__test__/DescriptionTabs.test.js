import React from "react";
import { createStore } from "redux";
import { render } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import DescriptionTabs from "../DescriptionTabs";

import { tempStore } from "./tempStore";

describe("DescriptionTabs component test", () => {
  test("create DescriptionTabs components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    const props = {
      index: 0,
      isForProperty: false,
    };
    const wrapper = render(
      <MemoryRouter>
        <DescriptionTabs {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });
});
