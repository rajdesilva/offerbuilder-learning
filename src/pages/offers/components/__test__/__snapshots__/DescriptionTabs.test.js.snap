// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionTabs component test create DescriptionTabs components snapshot 1`] = `
<div
  class="ant-tabs ant-tabs-top"
  data-testid="description-tab"
>
  <div
    class="ant-tabs-nav"
    role="tablist"
  >
    <div
      class="ant-tabs-nav-wrap"
    >
      <div
        class="ant-tabs-nav-list"
        style="transform: translate(0px, 0px);"
      >
        <div
          class="ant-tabs-tab ant-tabs-tab-active"
        >
          <div
            aria-controls="rc-tabs-test-panel-EN"
            aria-selected="true"
            class="ant-tabs-tab-btn"
            id="rc-tabs-test-tab-EN"
            role="tab"
            tabindex="0"
          >
            English
          </div>
        </div>
        <div
          class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
        />
      </div>
    </div>
    <div
      class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
    >
      <button
        aria-controls="rc-tabs-test-more-popup"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-hidden="true"
        class="ant-tabs-nav-more"
        id="rc-tabs-test-more"
        style="visibility: hidden; order: 1;"
        tabindex="-1"
        type="button"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="ant-tabs-content-holder"
  >
    <div
      class="ant-tabs-content ant-tabs-content-top"
    >
      <div
        aria-hidden="false"
        aria-labelledby="rc-tabs-test-tab-EN"
        class="ant-tabs-tabpane ant-tabs-tabpane-active"
        id="rc-tabs-test-panel-EN"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-4"
            style="padding: 4px 4px 4px 4px;"
          >
            Title For Customers
          </div>
          <div
            class="ant-col ant-col-8"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input"
                      data-testid="EN-offer-title"
                      id="EN_title"
                      maxlength="5000"
                      rows="1"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-4"
            style="padding: 4px 4px 4px 4px;"
          >
            Description
          </div>
          <div
            class="ant-col ant-col-16"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <textarea
                      class="ant-input"
                      data-testid="supplier-description-textarea"
                      id="EN_description"
                      maxlength="50000"
                      placeholder=""
                      rows="6"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-4"
            style="padding: 4px 4px 4px 4px;"
          >
            Short Description
          </div>
          <div
            class="ant-col ant-col-8"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input"
                      data-testid="supplier-desc-input"
                      id="EN_shortDescription"
                      maxlength="5000"
                      placeholder=""
                      rows="1"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-4"
            style="padding: 4px 4px 4px 4px;"
          >
            Terms And Conditions
          </div>
          <div
            class="ant-col ant-col-8"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <textarea
                      class="ant-input"
                      id="EN_termsAndConditions"
                      maxlength="10000"
                      rows="3"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
